collie/model/hybrid_matrix_factorization.py

```diff
@@ -439,20 +469,16 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
                 + self.item_biases(items).squeeze(1)
             )
         elif self.hparams.stage in ('metadata_only', 'all') and self.user_metadata is None:
-            # TODO: remove self.device and let lightning do it
-            item_metadata_output = self.item_metadata[items, :].to(self.device)
-            if self.item_metadata_layers is not None:
-                for metadata_nn_layer in self.item_metadata_layers:
-                    item_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(item_metadata_output)
-                        )
-                    )
+            self._calculate_metadata_output(
```

Comment:
I think my preference here would be to have `_calculate_metadata_output` return the actual metadata output, rather than setting it as a class attribute. 

```suggestion
            item_metadata_output = self._calculate_metadata_output(
```

What do you think?